nemo/lightning/pytorch/strategies.py

```diff
@@ -541,7 +542,24 @@ def load_model_state_dict(self, checkpoint: Mapping[str, Any], strict: bool = Tr
     @override
     def checkpoint_io(self) -> CheckpointIO:
         if self._checkpoint_io is None:
-            self._checkpoint_io = MegatronCheckpointIO()
+            checkpoint_callback = self.trainer.checkpoint_callback
+            async_save = getattr(checkpoint_callback, "async_save", False)
+            if async_save:
+                self._checkpoint_io = AsyncFinalizableCheckpointIO(
+                    MegatronCheckpointIO(
+                        save_ckpt_format='torch_dist',
```

Comment:
The `save_ckpt_format`,  `async_save` and `assume_constant_structure` should be configurable too - what's the plan for this?